### PR TITLE
Add extension method for IParameterModelConvention.

### PR DESCRIPTION
**Pull request by jbagga**
Addresses #5618 

cc @rynowak

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Core/DependencyInjection/ApplicationModelConventionExtensions.cs [hunk 1]
```diff
@@ -50,6 +50,60 @@ public static class ApplicationModelConventionExtensions
...
+        /// </summary>
+        /// <param name="conventions">The list of <see cref="IApplicationModelConvention"/>
+        /// in <see cref="AspNetCore.Mvc.MvcOptions"/>.</param>
+        /// <param name="parameterModelConvention">The <see cref="IParameterModelConvention"/> which needs to be
+        /// added.</param>
+        public static void Add(
+            this IList<IApplicationModelConvention> conventions,
+            IParameterModelConvention parameterModelConvention)
+        {
+            conventions.Add(new ParameterApplicationModelConvention(parameterModelConvention));
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Core/DependencyInjection/ApplicationModelConventionExtensions.cs]: null checks 😆

File: src/Microsoft.AspNetCore.Mvc.Core/DependencyInjection/ApplicationModelConventionExtensions.cs [hunk 2]
```diff
@@ -50,6 +50,60 @@ public static class ApplicationModelConventionExtensions
...
+
+        private class ParameterApplicationModelConvention : IApplicationModelConvention
+        {
+            private IParameterModelConvention _parameterModelConvention;
+
+            /// <summary>
+            /// Initializes a new instance of <see cref="ParameterApplicationModelConvention"/>.
+            /// </summary>
+            /// <param name="parameterModelConvention">The parameter convention to be applied on all parameters
+            /// in the application.</param>
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Core/DependencyInjection/ApplicationModelConventionExtensions.cs]: No need for docs on a private class.

File: test/Microsoft.AspNetCore.Mvc.Core.Test/DependencyInjection/ApplicationModelConventionExtensionsTest.cs [hunk 3]
```diff
@@ -13,6 +13,33 @@ namespace Microsoft.Extensions.DependencyInjection
...
+            options.Conventions[0].Apply(app);
+
+            // Assert
+            foreach (var controller in app.Controllers)
+            {
+                foreach (var action in controller.Actions)
+                {
+                    foreach (var parameter in action.Parameters)
+                    {
+                        Assert.True(parameter.Properties.ContainsKey("TestProperty"));
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/DependencyInjection/ApplicationModelConventionExtensionsTest.cs]: Try not to use `Assert.True` if possible. Use `Assert.Contains` instead. `Assert.True` is like the Guy Fieri of assertions

File: src/Microsoft.AspNetCore.Mvc.Core/DependencyInjection/ApplicationModelConventionExtensions.cs (hunk 1 above)
Comment (jbagga) [inline src/Microsoft.AspNetCore.Mvc.Core/DependencyInjection/ApplicationModelConventionExtensions.cs]: Should I also add null checks for the `Add` method with `IActionModelConvention` in this PR?

**Comment by jbagga** [comment]
@rynowak Updated

File: src/Microsoft.AspNetCore.Mvc.Core/DependencyInjection/ApplicationModelConventionExtensions.cs (hunk 1 above)
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Core/DependencyInjection/ApplicationModelConventionExtensions.cs]: Yeah, if they are missing, go ahead and add em

File: test/Microsoft.AspNetCore.Mvc.Core.Test/DependencyInjection/ApplicationModelConventionExtensionsTest.cs [hunk 4]
```diff
@@ -13,6 +13,34 @@ namespace Microsoft.Extensions.DependencyInjection
...
+
+            // Assert
+            foreach (var controller in app.Controllers)
+            {
+                foreach (var action in controller.Actions)
+                {
+                    foreach (var parameter in action.Parameters)
+                    {
+                        Assert.Contains("TestProperty", parameter.Properties.Keys);
+                        Assert.Contains("TestValue", parameter.Properties.Values);
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/DependencyInjection/ApplicationModelConventionExtensionsTest.cs]: btw **the** way to do this kind of thing is with `Assert.Collection` - learn it and 😍 it. This isn't really wrong, but `Assert.Collection` does exactly what you want. Or in this case `Assert.Single` would be great too.

```
Assert.Collection(parameter.Properties, kvp => kvp.Key == "..." && kvp.Value == "...");
```
**or**
```
var kvp = Assert.Single(parameter.Properties);
Assert.Equal("...", kvp.Key);
Assert.Equal("...", kvp.Value);
```

Comment (jbagga) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/DependencyInjection/ApplicationModelConventionExtensionsTest.cs]: Thanks! Is it okay if I change the other tests (DefaultActionModelConvention and DefaultControllerModelConvention) to be consistent with this? (Not use `Assert.True` and  follow the `Assert.Single` pattern?)

Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/DependencyInjection/ApplicationModelConventionExtensionsTest.cs]: Yeah that would be dope

**Pull request merged**